diff --git a/lib/posix-fdio/fdctl.c b/lib/posix-fdio/fdctl.c
index 04bc0ed..1cbb4b6 100644
--- a/lib/posix-fdio/fdctl.c
+++ b/lib/posix-fdio/fdctl.c
@@ -35,7 +35,6 @@ int uk_sys_ioctl(struct uk_ofile *of, int cmd, void *arg)
 			uk_or(&of->mode, O_NONBLOCK);
 		else
 			uk_and(&of->mode, ~O_NONBLOCK);
-		return 0;
 	}
 
 	iolock = _SHOULD_LOCK(of->mode);
diff --git a/lib/uknetdev/include/uk/netdev.h b/lib/uknetdev/include/uk/netdev.h
index 03bfe4a..5b3c46e 100644
--- a/lib/uknetdev/include/uk/netdev.h
+++ b/lib/uknetdev/include/uk/netdev.h
@@ -125,6 +125,24 @@ const char *uk_netdev_drv_name_get(struct uk_netdev *dev);
  */
 enum uk_netdev_state uk_netdev_state_get(struct uk_netdev *dev);
 
+/**
+ * Query current link state of the network device.
+ *
+ * @param dev
+ *   The Unikraft Network Device.
+ * @return
+ *   1 if link is up, 0 if link is down, or 1 if unhandled/unsupported by driver.
+ */
+static inline int uk_netdev_link_state_get(struct uk_netdev *dev)
+{
+	UK_ASSERT(dev);
+
+	if (!dev->ops || !dev->ops->link_state_get)
+		return 1;
+
+	return dev->ops->link_state_get(dev);
+}
+
 /**
  * Probes an unprobed Unikraft network device and turns it
  * into unconfigured state. After successful probing, the
diff --git a/lib/uknetdev/include/uk/netdev_core.h b/lib/uknetdev/include/uk/netdev_core.h
index 3309d11..3d2c915 100644
--- a/lib/uknetdev/include/uk/netdev_core.h
+++ b/lib/uknetdev/include/uk/netdev_core.h
@@ -394,6 +394,9 @@ typedef int (*uk_netdev_tx_one_t)(struct uk_netdev *dev,
 				  struct uk_netdev_tx_queue *queue,
 				  struct uk_netbuf *pkt);
 
+/** Driver callback type to query link state (1 = up, 0 = down). */
+typedef int (*uk_netdev_link_state_get_t)(struct uk_netdev *dev);
+
 /**
  * A structure containing the functions exported by a driver.
  */
@@ -426,6 +429,9 @@ struct uk_netdev_ops {
 	uk_netdev_txq_configure_t       txq_configure;
 	uk_netdev_rxq_configure_t       rxq_configure;
 	uk_netdev_start_t               start;
+
+	/** Link state query. */
+	uk_netdev_link_state_get_t      link_state_get;   /* optional */
 };
 
 /**
diff --git a/lib/uksched/exportsyms.uk b/lib/uksched/exportsyms.uk
index 47e6a7d..3b5de2b 100644
--- a/lib/uksched/exportsyms.uk
+++ b/lib/uksched/exportsyms.uk
@@ -40,3 +40,4 @@ __uk_sched_thread_current
 sched_yield
 sched_getaffinity
 sched_setaffinity
+uk_sched_head
diff --git a/plat/kvm/x86/link64.lds.S b/plat/kvm/x86/link64.lds.S
index fc156a0..5ad621f 100644
--- a/plat/kvm/x86/link64.lds.S
+++ b/plat/kvm/x86/link64.lds.S
@@ -37,8 +37,6 @@ PHDRS
 	text PT_LOAD FLAGS(PHDRS_PF_RX);
 	rodata PT_LOAD FLAGS(PHDRS_PF_R);
 	data PT_LOAD;
-	tls PT_TLS;
-	tls_load PT_LOAD;
 	stack PT_GNU_STACK FLAGS(PHDRS_PF_RW);
 }
 
